### ISSUE TITLE
TextureImporter fails while importing .bmp files

### ISSUE THREAD

**Issue by Tzenchor** [mono/MonoGame#4296]
TextureImporter is raising a NullReferenceException on line 236, when trying to import bitmaps with less than 8 bits per pixel.
This is happening in the develop branch. In version 3.4 it worked perfectly.

The reason is that the code is hardcoded to transform files with 8, 16, 24, or 32 bits per pixel, but does nothing with smaller sizes.

I have a fix for this ready that seems to work, at least for monochrome and 4 bits .bmp files, but I want to have a look to the rest of the code and possibly simplify it

**Comment by tomspilman** [comment]
> but does nothing with smaller sizes.

Just something that wasn't accounted for.

@KonajuGames ?

**Comment by KonajuGames** [comment]
Correct. Didn't account for anything less than 8-bits per pixel in BMP form.

If you have a fix, we'll be happy to have a look at it if you submit it as
a PR.

**Comment by Tzenchor** [comment]
I have found another problem while refactoring.
I have created a pair of bitmaps with 16 bits per pixel to test the other cases, and created the corresponding tests.
Well, the tests are failing because the test expects to receive data with SurfaceFormat Color, but it is getting bgra5551 and bgra565 respectively. And i'll get similar results with images having the RGBF and RGBAF format (floats instead of integers).
Should I preserve this results and modify the tests, or convert all of them to 32 bits like the other bitmaps?
